Changed img-cmd script to work with new Oldenborg dataset

chauvuha opened the pull request.

chauvuha committed:
Changed img-cmd script to work with new Oldenborg dataset

chauvuha committed:
Update img_cmd_training_new_dataset.py

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+from argparse import ArgumentParser
+
+import matplotlib.pyplot as plt
+import os.path
```
We probably want to use `pathlib` instead of `os.path`

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+# Constants 
+# Load the dataset artifact
+
+VALID_PCT = 0.05
```
These should probably all be program arguments.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+    "resnet34": resnet34
+}
+
+DATASET_DIR.ls()
```
Do you want to print this here? Or more likely just remove the line in this script since it is no longer a notebook.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+
+DATASET_DIR.ls()
+
+def filename_to_class(filename: str) -> str:
```
This reminds me: in `boxnav` we should always give the angle and not set it to zero since we handle that here.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+
+def filename_to_class(filename: str) -> str:
+    angle = float(filename.split("_")[1].split(".")[0].replace("p", "."))
+    if angle > 0:
```
Instead of `> 0` we should come up with a threshold.

chauvuha replied:
We are not quite sure we understand what you mean by threshold with the angle. Would you mind elaborating?

anthonyjclark replied:
This requires two changes:

* In `boxnav` we should use the angle in the filenames without any thresholding.
* Here we should say that any angle between, for example, -5 degrees and 5 degrees is given the label `forward.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+        # Opens image file and ensures dimension of channels included
+        img = Image.open(img_filename).convert('RGB')
+        # Resizes the image
+        img = img.resize((224, 224))
```
`224` should probably be a parameter of the constructor.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+        # Data and the label associated with that data
+        return (img, cmd), label
+
+class cmd_model(nn.Module):
```
```suggestion
class CommandModel(nn.Module):
```

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+        super(cmd_model, self).__init__()
+        self.cnn = arch(pretrained=pretrained)
+        
+        self.fc1 = nn.Linear(self.cnn.fc.out_features + 1, 512)
```
We probably want to make this part of the network less arbitrary. Make a note that we should try other architectures here.

anthonyjclark commented on img_cmd_training_new_dataset.py:
```diff
@@ -0,0 +1,280 @@
+        self.r1 = nn.ReLU(inplace=True)
+        self.fc2 = nn.Linear(512, 3)
+        
+    def forward(self, data):
```
This method needs some documentation so that we don’t forget what it is doing on each line.

chauvuha committed:
Rename Command model class for clarity

Co-authored-by: Anthony Clark <anthonyjclark@gmail.com>

chauvuha committed:
Update img_cmd_training_new_dataset.py

+ added parse args: valid_pct and img_size and fixed functions accordingly
+ changed os.path to pathlib
+ added documentation on cmd_model

chauvuha committed:
Convert valid_pct from string to float

chauvuha committed:
Convert img_size from string to float

anthonyjclark commented:
Work was completed elsewhere.
